<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <link href='https://fonts.googleapis.com/css?family=Architects+Daughter' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-light.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print">

    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->

    <title>Jamm by mahmoudibrahim</title>
  </head>

  <body>
    <header>
      <div class="inner">
        <h1>Jamm</h1>
        <h2>JAMM Peak Finder for Sequencing Datasets</h2>
        <a href="https://github.com/mahmoudibrahim/JAMM" class="button"><small>View project on</small> GitHub</a>
      </div>
    </header>

    <div id="content-wrapper">
      <div class="inner clearfix">
        <section id="main-content">
          <h1>
<a id="jamm-peak-finder" class="anchor" href="#jamm-peak-finder" aria-hidden="true"><span class="octicon octicon-link"></span></a>JAMM Peak Finder</h1>

<p><strong>JAMM is a peak finder for NGS datasets (ChIP-Seq, ATAC-Seq, DNase-Seq..etc.) that can integrate replicates and assign peak boundaries accurately. JAMM is applicable to both broad and narrow datasets. Read more <a href="http://bioinformatics.oxfordjournals.org/content/early/2014/09/15/bioinformatics.btu568">in JAMM's publication</a>.</strong> </p>

<p>JAMM was developed at the <a href="http://ohlerlab.mdc-berlin.net/">Ohler lab</a> at <a href="http://www.mdc-berlin.de/13800178/en/bimsb">BIMSB-MDC</a> in Berlin.</p>

<p>If you have questions or need help running JAMM please email us at <a href="http://www.google.com/recaptcha/mailhide/d?k=01vPijd2GG0LEbZV2NyE_rSA==&amp;c=49GEiFp47dZQV_120clczwxYcP3tQ98VWBJtNl6_dBw=">this email</a>, we will be happy to help you.</p>

<h2>
<a id="news-and-updates" class="anchor" href="#news-and-updates" aria-hidden="true"><span class="octicon octicon-link"></span></a>News and Updates</h2>

<ul>
<li><p><strong>Jul 30 2015:</strong> <em>New version of JAMM released (v1.0.7rev2)</em>. This version has some minor bug fixes and some cryptic errors replaced by user-friendly messages.</p></li>
<li><p><strong>Feb 18 2015:</strong> <em>New version of JAMM released (v1.0.7rev1)</em>. This version introduces clustering model initialization options among other things, optional PCR duplicates filtering and reports the seed used for all randomization/sampling steps.</p></li>
<li><p><strong>Jan 14 2015:</strong> New questions and answers have been added to the [<a href="https://code.google.com/p/jamm-peak-finder/wiki/FAQs">https://code.google.com/p/jamm-peak-finder/wiki/FAQs</a> FAQs wiki page]. Why does JAMM output a large number of peaks? What is the "filtered" list?...Those questions and more are answered in the [<a href="https://code.google.com/p/jamm-peak-finder/wiki/FAQs">https://code.google.com/p/jamm-peak-finder/wiki/FAQs</a> FAQs wiki page]!</p></li>
<li><p><strong>Dec 06 2014:</strong> <em>New version of JAMM released (v1.0.6rev3)</em>. This version fixes a bug related to bash localization that affects peak score sorting for some users. You can read more in the changelog. </p></li>
<li><p><strong>Nov 25 2014:</strong> <em>New version of JAMM released (v1.0.6rev2)</em>. This version fixes bugs in the previous released version JAMMv1.0.6rev1, that cause paired-end option to not function.</p></li>
<li><p><strong>Nov 11 2014:</strong> <em>New version of JAMM released (v1.0.6rev1)</em>. This version has two main changes: <em>NO dependencies on "sqldf" R package</em>, and <em>optional</em> window fold enrichment cutoff to reduce running time and get only more highly confident peaks.</p></li>
<li><p><strong>Nov 11 2014:</strong> After a recent R update, certain versions of some R packages are not compatible with certain versions of the "sqldf" R package. This affects all JAMM versions from v1.0.4 until v1.0.6. Starting v1.0.6rev1, JAMM has NO dependencies on "sqldf" package.</p></li>
</ul>

<p>===Download JAMM===
<em>Current Version:</em> JAMM v1.0.7rev2 ([<a href="https://drive.google.com/file/d/0B8nxBVNVchN9QXl0QzJ3Ql9IRDA/view?usp=sharing">https://drive.google.com/file/d/0B8nxBVNVchN9QXl0QzJ3Ql9IRDA/view?usp=sharing</a> Download])</p>

<p>Version used in publication: JAMM v1.0.4rev1 ([<a href="https://drive.google.com/file/d/0B8nxBVNVchN9eUtrQUU0RWV0bGc/edit?usp=sharing">https://drive.google.com/file/d/0B8nxBVNVchN9eUtrQUU0RWV0bGc/edit?usp=sharing</a> Download])</p>

<p>[<a href="http://code.google.com/p/jamm-peak-finder/wiki/Downloads">http://code.google.com/p/jamm-peak-finder/wiki/Downloads</a> Previous Versions and Changelog] - [<a href="https://drive.google.com/file/d/0B8nxBVNVchN9eUdJV1FzdTEyTVU/view?usp=sharing">https://drive.google.com/file/d/0B8nxBVNVchN9eUdJV1FzdTEyTVU/view?usp=sharing</a> Details on differences between all released JAMM versions]</p>

<p>===JAMM Documentation=== </p>

<ul>
<li>[<a href="http://code.google.com/p/jamm-peak-finder/wiki/IntroductionAndInstallation">http://code.google.com/p/jamm-peak-finder/wiki/IntroductionAndInstallation</a> Introduction and Installation]</li>
<li>[<a href="http://code.google.com/p/jamm-peak-finder/wiki/Usage">http://code.google.com/p/jamm-peak-finder/wiki/Usage</a> How to Use]</li>
<li>[<a href="http://code.google.com/p/jamm-peak-finder/wiki/FAQs">http://code.google.com/p/jamm-peak-finder/wiki/FAQs</a> Frequently Asked Questions]</li>
<li>Of course it is always good to read [<a href="http://bioinformatics.oxfordjournals.org/content/31/1/48">http://bioinformatics.oxfordjournals.org/content/31/1/48</a> the actual paper] :)</li>
</ul>

<p>Please check [<a href="https://code.google.com/p/jamm-peak-finder/w/list">https://code.google.com/p/jamm-peak-finder/w/list</a> other Wiki pages] for more information on JAMM. </p>

<p>If you still have questions or need help running JAMM please post to [<a href="https://groups.google.com/forum/#!forum/jamm-peak-finder">https://groups.google.com/forum/#!forum/jamm-peak-finder</a> JAMM's Google group], or if you prefer you can email us at [<a href="http://www.google.com/recaptcha/mailhide/d?k=01vPijd2GG0LEbZV2NyE_rSA==&amp;c=49GEiFp47dZQV_120clczwxYcP3tQ98VWBJtNl6_dBw=">http://www.google.com/recaptcha/mailhide/d?k=01vPijd2GG0LEbZV2NyE_rSA==&amp;c=49GEiFp47dZQV_120clczwxYcP3tQ98VWBJtNl6_dBw=</a> this email].</p>

<p>===Example Dataset===
<em>You can use [<a href="https://drive.google.com/file/d/0B8nxBVNVchN9cFFzQnQxMnNQUjQ/edit?usp=sharing">https://drive.google.com/file/d/0B8nxBVNVchN9cFFzQnQxMnNQUjQ/edit?usp=sharing</a> this small dataset] to test if JAMM is installed correctly (Installation instructions [<a href="https://code.google.com/p/jamm-peak-finder/wiki/IntroductionAndInstallation">https://code.google.com/p/jamm-peak-finder/wiki/IntroductionAndInstallation</a> here]).</em></p>

<p>This dataset is chromosome 21, in K562 CTCF ChIP-Seq from [<a href="https://genome.ucsc.edu/ENCODE/">https://genome.ucsc.edu/ENCODE/</a> ENCODE].</p>

<p>===Quick Tutorial=== 
Before going ahead with this tutorial, please make sure JAMM is running properly on your machine using the small dataset provided above.</p>

<p>The purpose of this tutorial is to help you get started with running JAMM. The tutorial uses ENCODE (Broad Institute) <code>H3K27ac</code> ChIP-Seq full dataset from <code>HeLa-S3</code> cells.</p>

<p>It starts with the aligned BED files which we provide here. Those files were generated by aligning reads to hg19 using Bowtie2, followed by PCR duplicates removal using SAMtools. </p>

<p><em>Step 1:</em> Download the <code>H3K27ac</code> ChIP-Seq BED files corresponding to Replicate 1, Replicate 2 and ChIP Input (Warning: Files are big, up to about 4Gb). </p>

<ul>
<li><p>[<a href="https://drive.google.com/file/d/0B8nxBVNVchN9c2lQQmZBSXZwQWM/edit?usp=sharing">https://drive.google.com/file/d/0B8nxBVNVchN9c2lQQmZBSXZwQWM/edit?usp=sharing</a> Replicate 1]</p></li>
<li><p>[<a href="https://drive.google.com/file/d/0B8nxBVNVchN9OUlPQlQwNDRyWkU/edit?usp=sharing">https://drive.google.com/file/d/0B8nxBVNVchN9OUlPQlQwNDRyWkU/edit?usp=sharing</a> Replicate 2]</p></li>
<li><p>[<a href="https://drive.google.com/file/d/0B8nxBVNVchN9WHpXRTdpY29hNDQ/edit?usp=sharing">https://drive.google.com/file/d/0B8nxBVNVchN9WHpXRTdpY29hNDQ/edit?usp=sharing</a> ChIP HeLa-S3 Broad Institute Input]</p></li>
</ul>

<p><em>Step 2:</em> Download the chromosome size file corresponding to hg19 genome build. This file is needed by JAMM to run.</p>

<ul>
<li>[<a href="https://drive.google.com/file/d/0B8nxBVNVchN9SGNsQXlrQTg3UTA/edit?usp=sharing">https://drive.google.com/file/d/0B8nxBVNVchN9SGNsQXlrQTg3UTA/edit?usp=sharing</a> hg19 chromosome size file]</li>
</ul>

<p><em>Step 3:</em> Create the directory structure you need to run JAMM. We recommend you set up a main directory called "<code>H3K27ac</code>" and 2 additional sub-directories called "sample" and "control":</p>

<p>{{{
mkdir H3K27ac</p>

<p>mkdir H3K27ac/sample</p>

<p>mkdir H3K27ac/ctrl
}}}</p>

<p><em>Step 4:</em> Copy the Replicate 1 and Replicate 2 BED files to the "<code>H3K27ac</code>/sample" directory. Copy the Input file to the "<code>H2K27ac</code>/ctrl" directory.</p>

<p><em>Step 5:</em> Copy the chromosome size file to the "<code>H3K27ac</code>" directory.</p>

<p><em>Step 6:</em> You are now ready to run JAMM:</p>

<p>{{{
bash /path/to/JAMM/folder/JAMM.sh -s /path/to/your/"H3K27ac/sample"/folder -g /path/to/your/chromosome/size/file -o /path/to/folder/with/your/main/"H3K27ac"/folder -c /path/to/folder/with/your/"H3K27ac/ctrl"/folder
}}}</p>

<p><em>Step 7</em>: Check the output. You will have two new folders in your main "<code>H3K27ac</code>" directory. One is called "peaks" containing your peak files and one is called "xcorr" containing results for cross-correlation analysis.</p>

<p><em>We already analyzed this dataset using JAMMv1.0.4 and you can get <a href="Provided%20just%20as%20an%20example,%20not%20to%20be%20used%20directly%20in%20publications">https://drive.google.com/file/d/0B8nxBVNVchN9c01KNFBXMm5VeFE/edit?usp=sharing our results here</a>.</em></p>

<p>More information about JAMM output is provided on the [<a href="https://code.google.com/p/jamm-peak-finder/wiki/Usage">https://code.google.com/p/jamm-peak-finder/wiki/Usage</a> Usage Wiki page]</p>

<p>===Example Peak Calls (Provided just as an example, not to be used directly in publications)===
If you want to check out some JAMM results, you can [<a href="https://drive.google.com/file/d/0B8nxBVNVchN9RlhRcl9nUWNfN0k/edit?usp=sharing">https://drive.google.com/file/d/0B8nxBVNVchN9RlhRcl9nUWNfN0k/edit?usp=sharing</a> download JAMM peak calls for all ENCODE histone modification ChIP-Seq datasets] that were produced for the HeLaS3 cell line. Note: peak calls produced using JAMMv1.0.4. Note: Provided just as an example, not to be used directly in publications.</p>

<p>===How to Cite JAMM?===
Please cite:</p>

<p>Ibrahim MM, Lacadie SA, Ohler U (2015). JAMM: A Peak Finder for Joint Analysis of NGS Replicates. <em>Bioinformatics</em>, 31(1): 48-55. doi: [<a href="http://bioinformatics.oxfordjournals.org/content/early/2014/09/15/bioinformatics.btu568">http://bioinformatics.oxfordjournals.org/content/early/2014/09/15/bioinformatics.btu568</a> 10.1093/bioinformatics/btu568]</p>

<hr>

<p>===  ===
<em>Hint:</em> While you run JAMM, you can also go on jammin...
===  ===
</p>
        </section>

        <aside id="sidebar">
          <a href="https://github.com/mahmoudibrahim/JAMM/zipball/master" class="button">
            <small>Download</small>
            .zip file
          </a>
          <a href="https://github.com/mahmoudibrahim/JAMM/tarball/master" class="button">
            <small>Download</small>
            .tar.gz file
          </a>

          <p class="repo-owner"><a href="https://github.com/mahmoudibrahim/JAMM"></a> is maintained by <a href="https://github.com/mahmoudibrahim">mahmoudibrahim</a>.</p>

          <p>This page was generated by <a href="https://pages.github.com">GitHub Pages</a> using the Architect theme by <a href="https://twitter.com/jasonlong">Jason Long</a>.</p>
        </aside>
      </div>
    </div>

  
  </body>
</html>
