JAMM Peak Finder
======

**JAMM is a peak finder for NGS datasets (ChIP-Seq, ATAC-Seq, DNase-Seq..etc.) that can integrate replicates and assign peak boundaries accurately. JAMM is applicable to both broad and narrow datasets. Read more [in JAMM's publication](http://bioinformatics.oxfordjournals.org/content/early/2014/09/15/bioinformatics.btu568).** 

JAMM was developed at the [Ohler lab](http://ohlerlab.mdc-berlin.net/) at [BIMSB-MDC](http://www.mdc-berlin.de/13800178/en/bimsb) in Berlin.


If you have questions or need help running JAMM please email us at [this email](http://www.google.com/recaptcha/mailhide/d?k=01vPijd2GG0LEbZV2NyE_rSA==&c=49GEiFp47dZQV_120clczwxYcP3tQ98VWBJtNl6_dBw=), we will be happy to help you.

News and Updates
------

* **Jul 30 2015:** *New version of JAMM released (v1.0.7rev2)*. This version has some minor bug fixes and some cryptic errors replaced by user-friendly messages.

* **Feb 18 2015:** *New version of JAMM released (v1.0.7rev1)*. This version introduces clustering model initialization options among other things, optional PCR duplicates filtering and reports the seed used for all randomization/sampling steps.

* **Jan 14 2015:** New questions and answers have been added to the [https://code.google.com/p/jamm-peak-finder/wiki/FAQs FAQs wiki page]. Why does JAMM output a large number of peaks? What is the "filtered" list?...Those questions and more are answered in the [https://code.google.com/p/jamm-peak-finder/wiki/FAQs FAQs wiki page]!

* **Dec 06 2014:** *New version of JAMM released (v1.0.6rev3)*. This version fixes a bug related to bash localization that affects peak score sorting for some users. You can read more in the changelog. 

* **Nov 25 2014:** *New version of JAMM released (v1.0.6rev2)*. This version fixes bugs in the previous released version JAMMv1.0.6rev1, that cause paired-end option to not function.

* **Nov 11 2014:** *New version of JAMM released (v1.0.6rev1)*. This version has two main changes: *NO dependencies on "sqldf" R package*, and _optional_ window fold enrichment cutoff to reduce running time and get only more highly confident peaks.

* **Nov 11 2014:** After a recent R update, certain versions of some R packages are not compatible with certain versions of the "sqldf" R package. This affects all JAMM versions from v1.0.4 until v1.0.6. Starting v1.0.6rev1, JAMM has NO dependencies on "sqldf" package.

===Download JAMM===
*Current Version:* JAMM v1.0.7rev2 ([https://drive.google.com/file/d/0B8nxBVNVchN9QXl0QzJ3Ql9IRDA/view?usp=sharing Download])

Version used in publication: JAMM v1.0.4rev1 ([https://drive.google.com/file/d/0B8nxBVNVchN9eUtrQUU0RWV0bGc/edit?usp=sharing Download])


[http://code.google.com/p/jamm-peak-finder/wiki/Downloads Previous Versions and Changelog] - [https://drive.google.com/file/d/0B8nxBVNVchN9eUdJV1FzdTEyTVU/view?usp=sharing Details on differences between all released JAMM versions]

===JAMM Documentation=== 

 * [http://code.google.com/p/jamm-peak-finder/wiki/IntroductionAndInstallation Introduction and Installation]
 * [http://code.google.com/p/jamm-peak-finder/wiki/Usage How to Use]
 * [http://code.google.com/p/jamm-peak-finder/wiki/FAQs Frequently Asked Questions]
 * Of course it is always good to read [http://bioinformatics.oxfordjournals.org/content/31/1/48 the actual paper] :)

Please check [https://code.google.com/p/jamm-peak-finder/w/list other Wiki pages] for more information on JAMM. 

If you still have questions or need help running JAMM please post to [https://groups.google.com/forum/#!forum/jamm-peak-finder JAMM's Google group], or if you prefer you can email us at [http://www.google.com/recaptcha/mailhide/d?k=01vPijd2GG0LEbZV2NyE_rSA==&c=49GEiFp47dZQV_120clczwxYcP3tQ98VWBJtNl6_dBw= this email].

===Example Dataset===
*You can use [https://drive.google.com/file/d/0B8nxBVNVchN9cFFzQnQxMnNQUjQ/edit?usp=sharing this small dataset] to test if JAMM is installed correctly (Installation instructions [https://code.google.com/p/jamm-peak-finder/wiki/IntroductionAndInstallation here]).*

This dataset is chromosome 21, in K562 CTCF ChIP-Seq from [https://genome.ucsc.edu/ENCODE/ ENCODE].


===Quick Tutorial=== 
Before going ahead with this tutorial, please make sure JAMM is running properly on your machine using the small dataset provided above.

The purpose of this tutorial is to help you get started with running JAMM. The tutorial uses ENCODE (Broad Institute) `H3K27ac` ChIP-Seq full dataset from `HeLa-S3` cells.

It starts with the aligned BED files which we provide here. Those files were generated by aligning reads to hg19 using Bowtie2, followed by PCR duplicates removal using SAMtools. 


*Step 1:* Download the `H3K27ac` ChIP-Seq BED files corresponding to Replicate 1, Replicate 2 and ChIP Input (Warning: Files are big, up to about 4Gb). 

  * [https://drive.google.com/file/d/0B8nxBVNVchN9c2lQQmZBSXZwQWM/edit?usp=sharing Replicate 1]
      
  * [https://drive.google.com/file/d/0B8nxBVNVchN9OUlPQlQwNDRyWkU/edit?usp=sharing Replicate 2]

  * [https://drive.google.com/file/d/0B8nxBVNVchN9WHpXRTdpY29hNDQ/edit?usp=sharing ChIP HeLa-S3 Broad Institute Input]

*Step 2:* Download the chromosome size file corresponding to hg19 genome build. This file is needed by JAMM to run.

  * [https://drive.google.com/file/d/0B8nxBVNVchN9SGNsQXlrQTg3UTA/edit?usp=sharing hg19 chromosome size file]

*Step 3:* Create the directory structure you need to run JAMM. We recommend you set up a main directory called "`H3K27ac`" and 2 additional sub-directories called "sample" and "control":

{{{
mkdir H3K27ac

mkdir H3K27ac/sample

mkdir H3K27ac/ctrl
}}}

*Step 4:* Copy the Replicate 1 and Replicate 2 BED files to the "`H3K27ac`/sample" directory. Copy the Input file to the "`H2K27ac`/ctrl" directory.

*Step 5:* Copy the chromosome size file to the "`H3K27ac`" directory.

*Step 6:* You are now ready to run JAMM:

{{{
bash /path/to/JAMM/folder/JAMM.sh -s /path/to/your/"H3K27ac/sample"/folder -g /path/to/your/chromosome/size/file -o /path/to/folder/with/your/main/"H3K27ac"/folder -c /path/to/folder/with/your/"H3K27ac/ctrl"/folder
}}}

*Step 7*: Check the output. You will have two new folders in your main "`H3K27ac`" directory. One is called "peaks" containing your peak files and one is called "xcorr" containing results for cross-correlation analysis.

*We already analyzed this dataset using JAMMv1.0.4 and you can get [https://drive.google.com/file/d/0B8nxBVNVchN9c01KNFBXMm5VeFE/edit?usp=sharing our results here] (Provided just as an example, not to be used directly in publications).*

More information about JAMM output is provided on the [https://code.google.com/p/jamm-peak-finder/wiki/Usage Usage Wiki page]


===Example Peak Calls (Provided just as an example, not to be used directly in publications)===
If you want to check out some JAMM results, you can [https://drive.google.com/file/d/0B8nxBVNVchN9RlhRcl9nUWNfN0k/edit?usp=sharing download JAMM peak calls for all ENCODE histone modification ChIP-Seq datasets] that were produced for the HeLaS3 cell line. Note: peak calls produced using JAMMv1.0.4. Note: Provided just as an example, not to be used directly in publications.


===How to Cite JAMM?===
Please cite:

Ibrahim MM, Lacadie SA, Ohler U (2015). JAMM: A Peak Finder for Joint Analysis of NGS Replicates. _Bioinformatics_, 31(1): 48-55. doi: [http://bioinformatics.oxfordjournals.org/content/early/2014/09/15/bioinformatics.btu568 10.1093/bioinformatics/btu568]


----
===  ===
*Hint:* While you run JAMM, you can also go on jammin...
===  ===
<wiki:video url="https://www.youtube.com/watch?v=HSs1HgM0Wos"/>
